docs/readmes/proposals/p024_magma_settlement_service.md

```diff
@@ -0,0 +1,281 @@
+
+#### 3.2.8 RAP-In procedure
+
+The home network operator validates the data in the TAP files to ensure that it conforms to the TAP standard and to the terms of the roaming agreement. If the received TAP file contains any errors, the home network operator can reject the entire file or only the incorrect call event detail records. The incorrect file or records are returned to the visited network operator in a Returned Account Procedure (RAP) file.
+RAP process is used to return rejected TAP files and records to the visited network operator for corrections. A RAP file contains the rejected TAP file or records and additional data about the error, such as the error code or the error cause. The visited network operator corrects the errors and sends the corrected TAP file back to the home network operator.
+
+### 3.3 Integration interfaces
+
+The following figure shows the integration interfaces between Magma domain and OCS /DCH.
+
+![Integration interfaces](https://user-images.githubusercontent.com/93994458/149514277-062e0834-cb8c-49af-b0e4-867c4664c694.png)
+
+#### 3.3.1 DIAMETER Gy interface specification
+
+Interface transport protocol: TCP/Diameter/DCCA.
+The 3GPP based Gy interface between Session Proxy and Session Purchaser to implement real-time quota management for the packet data service for the roaming subscribers. While this interface resembles Gy in almost every way, this is a non-standard use of the Gy protocol defined to allow accounting at the operator level and allow the session purchaser a control point to approve or deny sessions at initiation or at update points throughout a session.
+The Session Purchaser is the Diameter Credit Control server, which provides the online charging data to the Session Proxy based on volume quota.
+The connection between the Session Proxy (client) and Session Purchaser (server) is TCP based. There are a series of message exchanges to manage quota in real-time.
+
+#### 3.3.2 MME/SGW ↔︎ Helium Miner
+
+Session lifecycle events and usage reports should be delivered from each deployed AGW to the Network-Server (helium-handler), which uses that information to update a state channel. The Helium Miner container does such communication between hotspot(AGW) and Network Server. So, a trusted channel should be formed between MME/SGW and Miner to let a Miner container communicate with the Helium network on behalf of AGW. For having a trusted channel between AGW and Miner new component is introduced - a session-forwarder. Session-forwarder is a mediator between Magma-specific components deployed on AGW and a Helium Miner.
+To let Miner trust Session-forwarder, session-forwarder is deployed on the same AGW where software is signed and Linux Secure Boot is used to verify all software components being loaded. So, Miner can trust to all received session-related messages. Miner just does forward all message to helium-handler.
```

Comment:
there is no session concept on inbound roaming. We do not use sessiond.

To have session concept you will have to change current inbound roaming flow which is
```
Attach 					UE -> 	MME 
Create session Request				MME -> s8_poxy
Create session Response  			MME <- s8_proxy
create flow						MME -> ovs
Attach Accepted				UE <-	MME
```

You will need something like

```
Attach 					UE -> 	MME 
Create session Request				MME -> s8_poxy
Create session Response        			MME <- s8_proxy
create session					MME -> 	sessiond
create_session_purchaser			 -> 	sessiond -> session_proxy -> session_purchaser
create_session_purchaser_acctp.                  <- 	sessiond <- session_proxy <- session_purchaser
create flow						sessiond -> pipelined -> ovs
Attach Accepted				UE <-      MME <- 	sessiond
```

Reply:
ok, we'll investigate s8_proxy